### PR TITLE
Pr mbr/1835

### PR DESCRIPTION
**Pull request by neutronc**
Rewriting ShareSellingRound based on StockRound SellShares Class.
Adding two save games with various Bugs reported.
Redid 1835_Stoll_12 Save game.

### CODE REVIEW

File: src/main/java/net/sf/rails/game/financial/ShareSellingRound.java [hunk 1]
```diff
@@ -54,7 +58,7 @@ public void start(RoundFacade parentRound, Player sellingPlayer, int cashToRaise
         this.cashToRaise = IntegerState.create(this, "CashToRaise", cashToRaise);
 
         this.dumpOtherCompaniesAllowed = dumpOtherCompaniesAllowed;
-        log.debug("Forced selling, dumpOtherCompaniesAllowed = {}", dumpOtherCompaniesAllowed);
+        log.debug("Forced selling, dumpOtherCompaniesAllowed = " + dumpOtherCompaniesAllowed);
```
Comment (jklap) [inline src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:61]: Looks like some of the logging variable changes were reverted?

File: src/main/java/net/sf/rails/game/financial/ShareSellingRound.java [hunk 2]
```diff
@@ -149,87 +149,101 @@ public void setSellableShares() {
...
+                            // dumping requires that the total is in the possibleSharesToSell list and that shareSize == 1
+                            // multiple shares have to be sold separately
+                            if (shareSize == 1 && possibleSharesToSell.contains(i * shareSize)) {
+                                sellableShares.add(new SellShares(company, shareSize, i, price, 1));
+                            }
+                        } else {
+                            // ... no dumping: standard sell
+                            sellableShares.add(new SellShares(company, shareSize, i, price, 0));
+                        }
+                    } else if (dumpIsPossible && i * shareSize >= dumpThreshold) {
```
Comment (jklap) [inline src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:235]: Maybe extract out the `i * shareSize` math as it looks to be used in a couple of places?

File: src/main/java/net/sf/rails/game/financial/ShareSellingRound.java (hunk 1 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:61]: Sorry about that overlook

File: src/main/java/net/sf/rails/game/financial/ShareSellingRound.java (hunk 2 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:235]: next project :) you are right

**Comment by madoar** [comment]
The tests are failing

**Comment by madoar** [comment]
Am I right in assuming that the tests are fixed by #206?

File: src/main/java/net/sf/rails/game/financial/StockRound.java [hunk 3]
```diff
@@ -720,10 +717,10 @@ protected boolean processGameSpecificAction(PossibleAction action) {
...
      * Start a company by buying one or more shares (more applies to e.g. 1841)
      *
-     * @param player  The player that wants to start a company.
-     * @param company The company to start.
-     * @param price   The start (par) price (ignored if the price is fixed).
-     * @param shares  The number of shares to buy (can be more than 1 in e.g.
+     * @param playerName  The player that wants to start a company.
+     * @param action containing the company to start, the price(par) and the number of shares to buy.
+     * price   The start (par) price (ignored if the price is fixed).
+     * shares  The number of shares to buy (can be more than 1 in e.g.
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/financial/StockRound.java:723]: I think these comments should go to the variable definitions below

File: src/main/java/net/sf/rails/game/financial/StockRound.java [hunk 4]
```diff
@@ -1907,7 +1906,7 @@ public int maxAllowedNumberOfSharesToBuy(Player player,
                             : playerShareLimit;
         }
         int maxAllowed = (limit - player.getPortfolioModel().getShare(company)) / shareSize;
-        //        log.debug("MaxAllowedNumberOfSharesToBuy = " + maxAllowed + " for company =  " + company + " shareSize " + shareSize);
+               log.debug("MaxAllowedNumberOfSharesToBuy = " + maxAllowed + " for company =  " + company + " shareSize " + shareSize);
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/financial/StockRound.java:1909]: The indentation looks wrong

File: src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java [hunk 5]
```diff
@@ -201,32 +201,36 @@ protected void gameSpecificChecks (PortfolioModel boughtFrom,
...
+                            "All", GameManager_1835.HE_ID));
+                }
+                break;
+            case GameManager_1835.MS_ID:   // Mecklenburg/Schwerin
+                if (sharesInIPO == 40) {  // 60% sold: release Oldenburg
+                    releaseCompanyShares(companyManager.getPublicCompany(GameManager_1835.OL_ID));
+                    ReportBuffer.add(this, LocalText.getText("SharesReleased",
+                            "All", GameManager_1835.OL_ID));
+                }
+                break;
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java:233]: I like the `switch` :+1:

File: src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java [hunk 6]
```diff
@@ -278,5 +282,41 @@ protected void setPriority(String string) {
...
+                    dumpedPlayer.getId(),
+                    company.getId()));
+
+        }
+
+        // Transfer the sold certificates
+        Portfolio.moveAll(certsToSell, bankTo);
+    }
+
+    private void executePresidentTransferAfterDump(PublicCompany company, Set<PublicCertificate> certsToSell, Player newPresident, int presSharesToSell, Player oldPresident, BankPortfolio bankTo) {
```
Comment (madoar) [inline src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java:304]: Can you please add a short comment to this method describing what it is meant to do?

File: src/main/java/net/sf/rails/ui/swing/ORWindow.java [hunk 7]
```diff
@@ -172,24 +172,27 @@ public void componentResized(ComponentEvent e) {
...
-            Rectangle bounds = ws.getBounds(ORWindow.this);
-            if (bounds.x != -1 && bounds.y != -1) setLocation(bounds.getLocation());
-            if (bounds.width != -1 && bounds.height != -1) setSize(bounds.getSize());
-            ws.set(frame);
+            public void run() {
+                //rearrange layout only if no docking framework active
+                if (!isDockingFrameworkEnabled()) {
+                    pack();
+                } else {
+                    setSize(new Dimension(600, 500));
```
Comment (madoar) [inline src/main/java/net/sf/rails/ui/swing/ORWindow.java:182]: Long-term question:

Should we assume these dimensions to be fixed size? Or should we calculate the preferred size somehow dynamically?

File: src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java (hunk 6 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java:304]: raise an issue so i dont forget :)

File: src/main/java/net/sf/rails/game/financial/StockRound.java (hunk 4 above)
Comment (neutronc) [reply src/main/java/net/sf/rails/game/financial/StockRound.java:1909]: yes :(

**Pull request merged**